tests/System.Text.Primitives.Tests/Formatting/TimeTests.cs

```diff
@@ -51,6 +53,7 @@ public void RandomDateTimeTests(char format)
         static void TestDateTimeFormat(DateTime dt, char format, SymbolTable symbolTable)
         {
             var expected = dt.ToString(format.ToString(), CultureInfo.InvariantCulture);
+            if (format == 'r') expected = expected.ToLowerInvariant();
```

Comment:
@KrzysztofCwalina Yes, I understand that, what I was referring to:
```csharp
    string payLoad = string.Format(System.Globalization.CultureInfo.InvariantCulture, "{0}\n{1}\n{2}\n{3}\n{4}\n",  
            verb.ToLowerInvariant(),  
            resourceType.ToLowerInvariant(),  
            resourceLink,  
            date.ToLowerInvariant(),  
            ""  
    );  

    byte[] hashPayLoad = hmacSha256.ComputeHash(System.Text.Encoding.UTF8.GetBytes(payLoad));  
```
If the above piece is performance critical, you should use the incremental Hash function, as the http verbs and the type are static (can be static byte arrays), the line break is a char/byte etc. Then you don't need ``ToString("R")``, but ``ToFormattedBytes("R")``. If your use case is really driven by the snippet above, then you'll probably gain way more by actually bypassing string formatting (as you said) instead of adding a new string formatting function.